kong/init.lua

```diff
@@ -360,6 +360,9 @@ function Kong.init()
   assert(db:connect())
   assert(db.plugins:check_db_against_config(config.loaded_plugins))
 
+  local clustering = require "kong.clustering"
```

Comment:
Why is this being required here and not at the top of the module? As far as I can tell, requiring `kong.clustering` is side-effect free.

Reply:
I think https://github.com/Kong/kong/pull/5294/files#diff-f984694673247937d61992dca2dfa663R21 is the problem.

Reply:
ahh, didn't catch that sneaking among the `requires`. so _that_ line needs to be deferred, instead?

Reply:
Yes, I smell a lazy loading of `declarative.new_config(kong.configuration)` could solve the issue. I made a card for that.